### PR TITLE
Fix test purification bugs

### PR DESCRIPTION
**Pull request by LesterKim**
### Description

- Fixed purification test bugs: psi_minus_psi_minus case and `complex_array_equal`
- Restyled constant variables
- Replaced magic numbers and strings

### Stakeholder(s)
@Alex-Kolar
@xwu64 
@acbarnett 
@nikoaliro

### CODE REVIEW

File: tests/entanglement_management/test_purification.py [hunk 1]
```diff
@@ -598,23 +602,20 @@ def test_BBPSSW_psi_minus_psi_minus():
 
         if ep1.meas_res == 0:
             counter += 1
-            assert complex_array_equal(phi_plus, state)
+            assert complex_array_equal(psi_plus, state)
```
Comment (ghost) [inline tests/entanglement_management/test_purification.py:606]: This was the bug, I take it?

**Review by nikoaliro — APPROVED** [review]

File: tests/entanglement_management/test_purification.py (hunk 1 above)
Comment (LesterKim) [reply tests/entanglement_management/test_purification.py:606]: That was one of them. The other is the actual `def complex_array_equal`.

File: tests/entanglement_management/test_purification.py [hunk 2]
```diff
@@ -490,13 +495,13 @@ def test_BBPSSW_psi_plus_psi_plus():
...
         assert id(ket1) == id(ket2)
         assert kept1.qstate_key in ket1.keys and kept2.qstate_key in ket1.keys
-        state = correct_order(ket1.state, ket1.keys)
 
+        state = correct_order(ket1.state, ket1.keys)
+        assert complex_array_equal(psi_plus, state)
         if ep1.meas_res == 0:
             counter += 1
-            assert complex_array_equal(psi_plus, state)
         else:
```
Comment (ghost) [inline tests/entanglement_management/test_purification.py:503]:
```suggestion
```

File: tests/entanglement_management/test_purification.py [hunk 3]
```diff
@@ -490,13 +495,13 @@ def test_BBPSSW_psi_plus_psi_plus():
...
-            assert complex_array_equal(psi_plus, state)
+            pass
```
Comment (ghost) [inline tests/entanglement_management/test_purification.py:504]:
```suggestion
```

**Pull request merged**